languages/elixir/exercises/concept/charlists/.docs/after.md

```diff
@@ -0,0 +1,164 @@
+
+```elixir
+'ABC\0'
+# => [65, 66, 67, 0]
+
+[65, 66, 67, 0]
+# => [65, 66, 67, 0]
+```
+
+Printability can be checked with [`List.ascii_printable?`].
+
+```elixir
+List.ascii_printable?([65, 66, 67])
+# => true
+List.ascii_printable?([65, 66, 67, 0])
+# => false
+```
+
+Keep in mind that those are two different ways of _displaying_ the same data. The values are strictly equal.
+
+```elixir
+'ABC' === [65, 66, 67]
+# => true
+```
+
+When printing a list with [`IO.inspect`][io-inspect], you can use the [`:charlists`][inspect-opts] option to control how lists are printed.
+
+```elixir
+IO.inspect('ABC', charlists: :as_charlists)
+# => prints 'ABC'
+
+IO.inspect('ABC', charlists: :as_lists)
+# => prints [65, 66, 67]
+```
+
+## Code points
+
+You can prepend a character with `?` to get its code point.
+
+```elixir
+?A
+# => 65
+
+[?:, ?)]
+# => ':)'
+```
+
+We have been saying how charlists are lists of integers, not chars. While there is no _basic_ char type in Elixir, there is a [built-in `char` type][typespecs-built-in-types] defined as an integer in the range `0..0x10FFFF`.
+
+## Charlists vs strings
+
+Each value in a charlist is the Unicode code point of a character whereas in a string, the codepoints are encoded as UTF-8.
+
+```elixir
+IO.inspect('tschüss', charlists: :as_lists)
+# => prints [116, 115, 99, 104, 252, 115, 115]
+
+IO.inspect("tschüss", binaries: :as_binaries)
+# => prints <<116, 115, 99, 104, 195, 188, 115, 115>>
+```
+
+Note how `ü`, code point `252`, is encoded in UTF-8 as `195` and `188`.
+
+**In practice, charlists are rarely used**. Their main use case is interfacing with Erlang, in particular when using older libraries that do not accept binaries as arguments.
+
+When working with Elixir, **always use strings to store text**. The `String` module offers a wide choice of functions to process text, functions not available for charlists.
+
+Charlists can be converted to strings with [`to_string`][kernel-to-string].
+
+```elixir
+to_string('hello')
+# => "hello"
+```
+
+## `case`
+
+`case` is a control flow structure that allows us to compare a given value against many patterns. Clauses in a `case` statement are evaluated from top to bottom, until a match is found.
+
+In many cases, using `case` is interchangeable with defining multiple function clauses.
+
+```elixir
+# one function clause, multiple case clauses
+def age_group(age) do
+  case age do
+    0 -> 'infant'
+    age when age < 4 -> 'baby'
+    age when age < 13 -> 'child'
+    age when age < 18 -> 'teenager'
+    _ -> 'adult'
+  end
+end
+
+# multiple function clauses, no case
+def age_group(0), do: 'infant'
+def age_group(age) when age < 4, do: 'baby'
+def age_group(age) when age < 13, do: 'child'
+def age_group(age) when age < 18, do: 'teenager'
+def age_group(_), do: 'adult'
+```
+
+There are no strict rules for choosing one over the other. It's a matter of personal preference that usually depends on context.
```

Comment:
It's too bad something like this isn't written in elixir, because I think this is often the best reason to choose one over the other: [Code: Align the happy path to the left edge](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88)